### PR TITLE
Avoid putting usernames, actual paths, etc... in generated docs.

### PR DESCRIPTION
**Pull request by geeknoid**
The generated docs for istioctl currently include the expanded value of my $HOME directory, since I generated the docs. Tweak the help text so it's generic instead of having people's name/paths.

### CODE REVIEW

File: pilot/cmd/istioctl/main.go [hunk 1]
```diff
@@ -484,11 +482,8 @@ istioctl context-create --api-server http://127.0.0.1:8080
 func init() {
 	rootCmd.PersistentFlags().StringVarP(&platform, "platform", "p", kubePlatform,
 		"Istio host platform")
-	defaultKubeconfig := path.Join(homedir.HomeDir(), ".kube/config")
-	if v := os.Getenv("KUBECONFIG"); v != "" {
-		defaultKubeconfig = v
-	}
-	rootCmd.PersistentFlags().StringVarP(&kubeconfig, "kubeconfig", "c", defaultKubeconfig,
+
+	rootCmd.PersistentFlags().StringVarP(&kubeconfig, "kubeconfig", "c", "$KUBECONFIG else $HOME/.kube/config",
```
Comment (ayj) [inline pilot/cmd/istioctl/main.go]: This is a functional regression (see https://github.com/istio/istio/pull/2365). For the purposes of doc generation can you explicitly set KUBECONFIG to a non-empty value, i.e. `KUBECONFIG=.kube/config istioctl collaterall --all`?

**Comment by geeknoid** [comment]
​Oh, I thought this parameter was only used for text display. Yeah, I guess
this won't work.

I'd rather find a way to fix this in the specific tool (istioctl) rather
than putting a istioctl-specific hack in the doc generation logic. But I
can't find a way to do this...​

**Comment by ayj** [comment]
You could try moving the kubeconfig defaulting logic from `init()` to a `PersistentPreRunE` function that is shared by all commands _except_ collateral.

**Comment by geeknoid** [comment]
PTAL.

Is this what you had in mind?

**Comment by ldemailly** [comment]
fyi: edited to avoid the term "PII" which is a very charged/loaded term
(will also delete this comment)

**Comment by ldemailly** [comment]
you need to add `pilot/cmd/istioctl/istioctl` to .gitignore and remove it from this PR

can you show the output of `istioctl --help` ?

**Comment by geeknoid** [comment]
Whoops, sorry for the binary. I've removed it.

As for the help output:

Flags:
  -h, --help                             help for istioctl
  -i, --istioNamespace string            Istio system namespace (default
"istio-system")
  -c, --kubeconfig string                Kubernetes configuration file
(default "$KUBECONFIG else $HOME/.kube/config")

**Comment by ldemailly** [comment]
cool, I think that's actually much more useful than the current
```
  -c, --kubeconfig string                Kubernetes configuration file (default "/Users/ldemailly/.kube/config")
```
/lgtm

**Comment by geeknoid** [comment]
The only way to get rid of the glog stuff is for us to link Istio
components with istio/glog instead of the real glog code. I just need to
make time to make that happen.

File: pilot/cmd/istioctl/main.go [hunk 2]
```diff
@@ -481,14 +489,23 @@ istioctl context-create --api-server http://127.0.0.1:8080
 	}
 )
 
+const defaultKubeConfigText = "$KUBECONFIG else $HOME/.kube/config"
```
Comment (ayj) [inline pilot/cmd/istioctl/main.go]: This isn't completely accurate on windows. You'd need to include $HOMEDIR, $HOMEPATH, and $USERPROFILE as well.

Comment (ayj) [reply pilot/cmd/istioctl/main.go]: That said, https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/#explore-the-homekube-directory assumes `$HOME` and the non-windows case so maybe its fine for doc purposes.

Comment (ldemailly) [reply pilot/cmd/istioctl/main.go]: yes I think it's incrementally better in this PR for most users and just nice that it also works on windows

**Comment by ayj** [comment]
> Is this what you had in mind?

Yup. `register`, `unregister`, and `kube-inject` also use `kubeconfig`.

**Comment by ldemailly** [comment]
/test e2e-smoke
(or you can just merge it...)

**Comment by ldemailly** [comment]
I apologize I didn't realize the issue in jason's comment wasn't cosmetic and we broke all presubmit for 4h as a result, until I merged https://github.com/istio/istio/pull/3126 - many thanks to @rkpagadala for alerting me of the issue

**Comment by costinm** [comment]
It seems this broke e2e-simple in CI ? 

"2018-02-03T04:34:08.539140Z	info	Running command /go/out/linux_amd64/release/istioctl kube-inject -f /tmp/istio.e2e.223059957/servicesToBeInjected.yaml413789679.yaml -o /tmp/istio.e2e.223059957/KubeInject222952578.yaml --hub istio --tag 48fd343d39d9400a5fe4374f76ac16566025ddfa -n istio-system -i istio-system --meshConfigMapName=istio

2018-02-03T04:34:09.164049Z	info	Command output: 
 Error: kubernetes configuration file "$KUBECONFIG else $HOME/.kube/config" does not exist
, err: exit status 255

File: pilot/cmd/istioctl/main.go (hunk 2 above)
Comment (costinm) [reply pilot/cmd/istioctl/main.go]: It'll be real fun for i18n...

**Comment by ldemailly** [comment]
rebase past 3126?

**Pull request merged**